tests/unit/lms/services/jstor/service_test.py

```diff
@@ -1,17 +1,23 @@
 from datetime import timedelta
 from functools import partial
-from unittest.mock import sentinel
+from unittest.mock import patch, sentinel
 
 import pytest
 
 from lms.services import ExternalRequestError
 from lms.services.jstor.service import ArticleNotFound, JSTORService
 from tests import factories
 
-API_URL = "http://api.jstor.org"
+API_URL = "http://jstore_api.example.com"
 
 
 class TestJSTORService:
+    def test_it_sets_tracking_headers(self, get_service):
+        svc = get_service(headers={"X-Header": "value"})
+
+        # pylint: disable=protected-access
+        assert svc._http.session.headers == {"X-Header": "value"}
```

Comment:
Patching private properties is unfortunate as an innocuous refactoring such as renaming the field can break the tests. Other fixtures in this module rely instead on patching the imports. I think that would look something like this:

```diff
diff --git a/tests/unit/lms/services/jstor/service_test.py b/tests/unit/lms/services/jstor/service_test.py
index 9ed4c6df..782ac25c 100644
--- a/tests/unit/lms/services/jstor/service_test.py
+++ b/tests/unit/lms/services/jstor/service_test.py
@@ -12,11 +12,10 @@ API_URL = "http://jstore_api.example.com"
 
 
 class TestJSTORService:
-    def test_it_sets_tracking_headers(self, get_service):
+    def test_it_sets_tracking_headers(self, get_service, http_service):
         svc = get_service(headers={"X-Header": "value"})
 
-        # pylint: disable=protected-access
-        assert svc._http.session.headers == {"X-Header": "value"}
+        assert http_service.session.headers == {"X-Header": "value"}
 
     @pytest.mark.parametrize("enabled", (True, False, None))
     @pytest.mark.parametrize("site_code", ("code", None, ""))
@@ -187,10 +186,13 @@ class TestJSTORService:
     def svc(self, get_service):
         return get_service()
 
+    @pytest.fixture
+    def HTTPService(self, patch):
+        return patch("lms.services.jstor.service.HTTPService")
+
     @pytest.fixture(autouse=True)
-    def http_service(self, svc):
-        with patch.object(svc, "_http") as http_service:
-            yield http_service
+    def http_service(self, HTTPService):
+        return HTTPService.return_value
 
     @pytest.fixture
     def ArticleMetadata(self, patch):
```

Reply:
Done!